uploader/s3.go

```diff
@@ -90,24 +93,43 @@ func (u *S3Uploader) Upload(ctx context.Context, local string, remote string, ct
 	return u.uploadFile(ctx, f, remote, contentType)
 }
 
+// CountingReader is a wrapper of io.Reader to count number of bytes read
+type CountingReader struct {
+	reader    io.Reader
+	BytesRead int
+}
+
+// Read aggregates number of bytes read
+func (r *CountingReader) Read(p []byte) (n int, err error) {
+	n, err = r.reader.Read(p)
+	if err == nil {
+		r.BytesRead += n
+	}
+	return
+}
+
 // UploadFile writes a single file only to S3!
 func (u *S3Uploader) uploadFile(ctx context.Context, local io.Reader, remote string, contentType string) error {
 	u.log.Printf("Attempting to upload file from: %s to: %s", local, path.Join(u.bucketName, remote))
 	if contentType == "" {
 		contentType = defaultS3ContentType
 	}
 
+	// wrap input io.Reader with a counting reader
+	reader := &CountingReader{local, 0}
```

Comment:
In general, it's better to initialize with explicit field names. Also, by default in Go, uninitialized fields have default values. For ints, this is 0.